plugins/module_utils/network/sonic/facts/sFlow/sFlow.py

```diff
@@ -0,0 +1,126 @@
+            that is formatted like argspec for this module. Can have empty values in it
+            :rtype: dictionary
+            :returns: dictionary that has options in same format as defined in argspec.
+            format looks something like: {"config": {"enabled":false, "interfaces":[{"name":"Etnernet0",
+            enabled: None...}], ...}}'''
+        formatted_data = {"config":{}}
+
+        formatted_data["config"]["agent"] = deepcopy(data["config"].get("agent", None))
+        formatted_data["config"]["enabled"] = deepcopy(data["config"].get("enabled", None))
+        formatted_data["config"]["polling_interval"] = deepcopy(data["config"].get("polling-interval", None))
+        
+        if "interfaces" in data:
+            formatted_data["config"]["interfaces"] = []
+            for interface in data["interfaces"]["interface"]:
+                if "config" in interface:
+                    formatted_interface = {}
+                    formatted_interface["name"] = deepcopy(interface.get("name", None))
+                    formatted_interface["enabled"] = deepcopy(interface["config"].get("enabled", None))
+                    formatted_interface["sampling_rate"] = deepcopy(interface["config"].get("sampling-rate", None))
+                    formatted_data["config"]["interfaces"].append(formatted_interface)
+
+        if "collectors" in data:
+            formatted_data["config"]["collectors"] = []
+            for collector in data["collectors"]["collector"]:
+                if "config" in collector:
+                    formatted_collector = {}
+                    formatted_collector["address"] = deepcopy(collector.get("address", None))
+                    formatted_collector["network_instance"] = deepcopy(collector["config"].get("network-instance", None))
+                    formatted_collector["port"] = deepcopy(collector["config"].get("port", None))
+                    formatted_data["config"]["collectors"].append(formatted_collector)
+
+        return formatted_data
+
+
+    def get_sFlow_info(self):
+        '''get the top level sFlow configuration on device
+        :rtype: dictionary
+        :returns: everything listed in resource's config
+        '''
+        uri_path = "data/openconfig-sampling-sflow:sampling/sflow"
+        method = "GET"
+        request = [{"path": uri_path, "method": method}]
+        # facts get request returns a dictionary, key to get facts data we care about
+        response_key = 'openconfig-sampling-sflow:sflow'
+
+        try:
+            response = edit_config(self._module, to_request(self._module, request))
+        except ConnectionError as exc:
+            self._module.fail_json(msg = str(exc))
+
+        response_body = {}
+        try:
+            response_body = response[0][1][response_key]
+        except Exception as e:
+            raise Exception("response from getting sFlow facts not formed as expected")
```

Comment:
`e` went unused here. Wouldn't the linter complain about this?

Reply:
don't remember any warnings, weirdly. removed in next commit